diff --git a/asm/macros/event.inc b/asm/macros/event.inc
index fdf86bc998..9fe9f02ddc 100644
--- a/asm/macros/event.inc
+++ b/asm/macros/event.inc
@@ -2933,6 +2933,25 @@
 	call EventScript_BrailleCursorWaitButton
 	.endm
 
+	.macro setwildbossbattle species:req, level:req, item=ITEM_NONE, move1_1=MOVE_NONE, move1_2=MOVE_NONE, move1_3=MOVE_NONE, move1_4=MOVE_NONE, species2=SPECIES_NONE, level2=0, item2=ITEM_NONE, move2_1=MOVE_NONE, move2_2=MOVE_NONE, move2_3=MOVE_NONE, move2_4=MOVE_NONE
+	callnative ScrCmd_setwildbossbattle
+	.2byte \species
+	.byte  \level
+	.2byte \item
+	.2byte \move1_1
+	.2byte \move1_2
+	.2byte \move1_3
+	.2byte \move1_4
+
+	.2byte \species2
+	.byte  \level2
+	.2byte \item2
+	.2byte \move2_1
+	.2byte \move2_2
+	.2byte \move2_3
+	.2byte \move2_4
+	.endm
+
 	@ Checks if any Pokemon in the player's party knows the given move.
 	@ Sets VAR_RESULT to the (zero-indexed) slot number, or PARTY_SIZE if none.
 	.macro checkpartymove move:req
diff --git a/include/script_pokemon_util.h b/include/script_pokemon_util.h
index 2b065a16a5..e17432308b 100644
--- a/include/script_pokemon_util.h
+++ b/include/script_pokemon_util.h
@@ -5,6 +5,10 @@ u32 ScriptGiveMon(u16 species, u8 level, enum Item item);
 u8 ScriptGiveEgg(u16 species);
 void CreateScriptedWildMon(u16 species, u8 level, enum Item item);
 void CreateShinyScriptedMon(u16 species, u8 level, enum Item item);
+void CreateScriptedWildBossMon(u16 species, u8 level, enum Item item, enum Move moves[MAX_MON_MOVES]);
+void CreateScriptedDoubleWildBossMon(u16 species1, u8 level1, enum Item item1, enum Move moves1[MAX_MON_MOVES], \
+    u16 species2, u8 level2, enum Item item2, \
+    enum Move moves2[MAX_MON_MOVES]);
 void CreateScriptedDoubleWildMon(u16 species, u8 level, enum Item item, u16 species2, u8 level2, enum Item item2);
 void ScriptSetMonMoveSlot(u8 monIndex, enum Move move, u8 slot);
 void ReducePlayerPartyToSelectedMons(void);
diff --git a/src/scrcmd.c b/src/scrcmd.c
index 54efff610d..7bf435c5e9 100644
--- a/src/scrcmd.c
+++ b/src/scrcmd.c
@@ -3210,6 +3210,38 @@ bool8 ScrCmd_setwildbattle(struct ScriptContext *ctx)
     return FALSE;
 }
 
+bool8 ScrCmd_setwildbossbattle(struct ScriptContext* ctx)
+{
+    u16 species = ScriptReadHalfword(ctx);
+    u8 level = ScriptReadByte(ctx);
+    enum Item item = ScriptReadHalfword(ctx);
+    enum Move moves[MAX_MON_MOVES];
+    for (u32 i = 0; i < MAX_MON_MOVES; i++)
+        moves[i] = ScriptReadHalfword(ctx);
+
+    u16 species2 = ScriptReadHalfword(ctx);
+    u8 level2 = ScriptReadByte(ctx);
+    enum Item item2 = ScriptReadHalfword(ctx);
+    enum Move moves2[MAX_MON_MOVES];
+    for (u32 i = 0; i < MAX_MON_MOVES; i++)
+        moves2[i] = ScriptReadHalfword(ctx);
+
+    Script_RequestEffects(SCREFF_V1);
+
+    if (species2 == SPECIES_NONE)
+    {
+        CreateScriptedWildBossMon(species, level, item, moves);
+        sIsScriptedWildDouble = FALSE;
+    }
+    else
+    {
+        CreateScriptedDoubleWildBossMon(species, level, item, moves, species2, level2, item2, moves2);
+        sIsScriptedWildDouble = TRUE;
+    }
+
+    return FALSE;
+}
+
 bool8 ScrCmd_dowildbattle(struct ScriptContext *ctx)
 {
     Script_RequestEffects(SCREFF_V1 | SCREFF_HARDWARE);
diff --git a/src/script_pokemon_util.c b/src/script_pokemon_util.c
index 42f7480b26..8faa276afb 100644
--- a/src/script_pokemon_util.c
+++ b/src/script_pokemon_util.c
@@ -3,6 +3,9 @@
 #include "battle_gfx_sfx_util.h"
 #include "berry.h"
 #include "caps.h"
+#include "constants/battle.h"
+#include "constants/global.h"
+#include "constants/moves.h"
 #include "data.h"
 #include "daycare.h"
 #include "decompress.h"
@@ -190,6 +193,94 @@ void CreateShinyScriptedMon(u16 species, u8 level, enum Item item)
     SetNuzlockeChecks();
 }
 
+void CreateScriptedWildBossMon(u16 species, u8 level, enum Item item, enum Move moves[MAX_MON_MOVES])
+{
+    u8 heldItem[2];
+
+    ZeroEnemyPartyMons();
+    u32 personality = GetMonPersonality(species, GetSynchronizedGender(STATIC_WILDMON_ORIGIN, species),
+                                        GetSynchronizedNature(STATIC_WILDMON_ORIGIN, species), RANDOM_UNOWN_LETTER);
+
+    struct Pokemon *mon = &gEnemyParty[0];
+
+    CreateMonWithIVs(mon, species, level, personality, OTID_STRUCT_PLAYER_ID, USE_RANDOM_IVS);
+
+    if (moves[0] == MOVE_NONE) {
+        GiveBoxMonInitialMoveset(&mon->box);
+    }
+    else {
+
+        for (u32 i = 0; i < MAX_MON_MOVES; i++) {
+            SetBoxMonData(&mon->box, MON_DATA_MOVE1 + i, &moves[i]);
+            u32 pp = GetMovePP(moves[i]);
+            SetBoxMonData(&mon->box, MON_DATA_PP1 + i, &pp);
+        }
+    }
+    if (item) {
+        heldItem[0] = item;
+        heldItem[1] = item >> 8;
+        SetMonData(&gEnemyParty[0], MON_DATA_HELD_ITEM, heldItem);
+    }
+}
+
+void CreateScriptedDoubleWildBossMon(u16 species1, u8 level1, enum Item item1, enum Move moves1[MAX_MON_MOVES],
+                                     u16 species2, u8 level2, enum Item item2, enum Move moves2[MAX_MON_MOVES])
+{
+    u8 heldItem1[2];
+    u8 heldItem2[2];
+    struct Pokemon *mon;
+    u32 personality;
+    ZeroEnemyPartyMons();
+
+    personality = GetMonPersonality(species1, GetSynchronizedGender(STATIC_WILDMON_ORIGIN, species1),
+                                    GetSynchronizedNature(STATIC_WILDMON_ORIGIN, species1), RANDOM_UNOWN_LETTER);
+
+    mon = &gEnemyParty[0];
+    CreateMonWithIVs(mon, species1, level1, personality, OTID_STRUCT_PLAYER_ID, USE_RANDOM_IVS);
+
+    if (moves1[0] == MOVE_NONE) {
+        GiveBoxMonInitialMoveset(&mon->box);
+    }
+    else {
+
+        for (u32 i = 0; i < MAX_MON_MOVES; i++) {
+            SetBoxMonData(&mon->box, MON_DATA_MOVE1 + i, &moves1[i]);
+            u32 pp = GetMovePP(moves1[i]);
+            SetBoxMonData(&mon->box, MON_DATA_PP1 + i, &pp);
+        }
+    }
+
+    if (item1) {
+        heldItem1[0] = item1;
+        heldItem1[1] = item1 >> 8;
+        SetMonData(mon, MON_DATA_HELD_ITEM, heldItem1);
+    }
+
+    personality = GetMonPersonality(species2, GetSynchronizedGender(STATIC_WILDMON_ORIGIN, species2),
+                                    GetSynchronizedNature(STATIC_WILDMON_ORIGIN, species2), RANDOM_UNOWN_LETTER);
+
+    mon = &gEnemyParty[1];
+    CreateMonWithIVs(mon, species2, level2, personality, OTID_STRUCT_PLAYER_ID, USE_RANDOM_IVS);
+
+    if (moves2[0] == MOVE_NONE) {
+        GiveBoxMonInitialMoveset(&mon->box);
+    }
+    else {
+
+        for (u32 i = 0; i < MAX_MON_MOVES; i++) {
+            SetBoxMonData(&mon->box, MON_DATA_MOVE1 + i, &moves2[i]);
+            u32 pp = GetMovePP(moves2[i]);
+            SetBoxMonData(&mon->box, MON_DATA_PP1 + i, &pp);
+        }
+    }
+
+    if (item2) {
+        heldItem2[0] = item2;
+        heldItem2[1] = item2 >> 8;
+        SetMonData(mon, MON_DATA_HELD_ITEM, heldItem2);
+    }
+}
+
 void CreateScriptedDoubleWildMon(u16 species1, u8 level1, enum Item item1, u16 species2, u8 level2, enum Item item2)
 {
     u8 heldItem1[2];
